### PR TITLE
Pre-C++17: Upgrade googletest from 1.7.0 to 1.8.x (2019-01-30)

### PR DESCRIPTION
**Pull request by uklotzde**
Upgrade to the latest v1.8.x version of googletest. Upgrading to 1.10.0 or master didn't work as expected.

3 remaining warnings during OSX/clang/SCons build on Travis:

```
ld: warning: directory not found for option '-Losx64_build/lib/googletest-1.8.x/googletest'
...
[LD] osx64_build/mixxx-test
ld: warning: directory not found for option '-Losx64_build/lib/googletest-1.8.x/googletest'
ld: warning: directory not found for option '-Losx64_build/lib/googletest-1.8.x/googletest'
Copy("mixxx-test", "osx64_build/mixxx-test")
scons: done building targets.
```

Note: Maintaining the SCons build takes more and more time. The CMake build works much better.

### CODE REVIEW

**Comment by uklotzde** [comment]
Ready!

File: CMakeLists.txt [hunk 1]
```diff
@@ -880,6 +880,20 @@ include(CPack)
 #
 # Tests
 #
+
+# CMake + googletest: https://crascit.com/2015/07/25/cmake-gtest/
+
+# Prevent GoogleTest from overriding our compiler/linker options
+# when building with Visual Studio
+set(gtest_force_shared_crt ON CACHE BOOL "" FORCE)
```
Comment (Holzhaus) [inline CMakeLists.txt]:
```suggestion
set(gtest_force_shared_crt ON CACHE BOOL "Pass Mixxx compiler/linker options to GoogleTest" FORCE)
```

**Review by Holzhaus — COMMENTED** [review]
Thanks, the cmake code looks good and is much less hacky than the previous implementation. I'm on mobile and didn't build mixxx myself yet though. I'll check this tomorrow.

**Comment by Holzhaus** [comment]
OK, this is what I tested:
- `cmake -DCMAKE_BUILD_TYPE=Release .. && cmake --build . -j18`
- `cmake -DCMAKE_BUILD_TYPE=Debug .. && cmake --build . -j18`
- `scons -j18`

Building and tests works in all configurations.

**Review by Holzhaus — APPROVED** [review]
LGTM. Great that this also fixes the travis builds.

**Review by daschuer — APPROVED** [review]

**Comment by daschuer** [comment]
Works here as well. Thank you.

**Pull request merged**